Tiltfile

```diff
@@ -153,6 +174,30 @@ def load_provider_tiltfiles():
                 provider_config["go_main"] = "main.go"
             providers[provider_name] = provider_config
 
+# load_addon_tiltfiles looks for tilt-addon.yaml files in the repositories listed in "addon_repos" in tilt settings and loads their config.
+def load_addon_tiltfiles():
+    addon_repos = settings.get("addon_repos", [])
+    for repo in addon_repos:
+        file = repo + "/tilt-addon.yaml" if os.path.exists(repo + "/tilt-addon.yaml") else repo + "/tilt-addon.json"
+        if not os.path.exists(file):
+            fail("Failed to load provider. No tilt-provider.{yaml|json} file found in " + repo)
+        addon_details = read_yaml(file, default = {})
+        if type(addon_details) != type([]):
+            addon_details = [addon_details]
+        for item in addon_details:
+            addon_name = item["name"]
+            addon_config = item["config"]
+            addon_config["debug"] = item["debug"]
```

Comment:
Why do we need this line for addons while we didn't need it for providers?

Reply:
Okay just realized why, shouldn't the debug config come from our tilt-settings.yaml instead? Like the debug config of providers?

Reply:
I think it's better to have all of the configuration for addons in the addons settings (honestly think it's the same for providers 🙂 ), it feels like it belongs there, but I can change it if you'd prefer.

Reply:
The nice thing is that tilt-settings.yaml is my configuration where I can decide locally how to run the dev env.

The other files are all part of the repository, which means I have to change the files under version control to change the config. Example: to add debug for the test extension I have to change the hard-coded config for test extension in the Tiltfile